sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +783,54 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(self,
+              host: MultihostHost,
+              domain: str = None,
+              proxy: str = None,
+              provider: str = None,
+              role: MultihostRole = None):
+        """
+        :param host: Multihost host.
+        :type host: MultihostHost
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        :param proxy: Ldap or files.
+        :type host: str
+        :param provider: Sssd providers
+        :type provider: Sssd providers (id, auth, chpass, ...)
+        :param role: MultihostRole
+        :type role: MultihostRole
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+
+        match proxy:
+            case "ldap":
+                self.sssd.fs.write('/etc/pam.d/sssdproxyldap', '''
+                        auth    required pam_ldap.so
+                        account required pam_ldap.so
+                        password required pam_ldap.so
+                        session required pam_ldap.so
+                        ''')
+                self.sssd.fs.write('/etc/nslcd.conf', f'uid nslcd\ngid ldap\nuri '
+                                                 f'ldap://{role.host.hostname}\n', dedent=False)
+                self.sssd.config[f'domain/{domain}'] = {
```

Comment:
`self.sssd.dom(domain).update(options)`

You need to update sssd configuration for both ldap and files so this can go after the match statement. Inside it, you just need to set the options.

Reply:
You didn't update the code, just moved it.